### PR TITLE
Add solid_mechanics and large strain versions of the bridge example

### LINKED ISSUE #5173: Port bridge example from TensorMechanics to SolidMechanics
As preparations for merging the two systems it would be nice to have an example to directly compare the two systems.
Assign to me.

### PR DESCRIPTION
**Pull request by dschwen**
Together with @bwspenc we created a large strain version of the bridge example and converted both versions over to solid_machanics to have a reference system for performance/memory comparison.
### Small strain

Total run time (1 cpu, macbook pro):
- tensor_mechanics: **35.5s**
- solid_mechanics: **45.1s**

Further analysis with Instruments shows that solid_mechanics uses stateful material properties without needing them in the model, while tensor_mechanics avoids this.
### Large strain

tensor_mechanics and solid_mechanics are head to head here. At a total run time of **~50s** solid_mechanics is about **1%** faster.

Memory use is very similar in both examples (slightly higher peak in solid_mechanics, _and_ larger time integral over the memory usage).
memory usage graph

In conclusion it looks like we do **not** need to spend time on making TM faster (i.e. making tensors symmetry aware). The resulting code would be a complicated monster and as a result we may loose the speed that we get from having trivially vectorizable loops.

Closes #5173

Attn.: @tonkmr, @sapitts, @cpritam, @WilkAndy, @amjokisaari

### CODE REVIEW

**Comment by dschwen** [comment]
P.S.: exodiffing the resulting exodus files shows negligible differences.

**Comment by amjokisaari** [comment]
Thanks for testing and sharing the results!  Loop vectorization -> speedup
was one reason I wrote the tensor implementations that way, and the other
was for ease of understanding what's going on under the hood.  I am glad to
see TM in all its versions is performing well :)
